src/main/java/io/lakefs/routerfs/PathMapper.java

```diff
@@ -233,7 +227,7 @@ public PathMapping(MappingConfig srcConfig, MappingConfig dstConfig, boolean def
             }
             this.fromScheme = srcConfig.getFromScheme();
             if (!defaultMapping) {
-                if (srcConfig.getPriority() != dstConfig.getPriority()) {
+                if (srcConfig.getPriority().intValue() != dstConfig.getPriority().intValue()) {
```

Comment:
The fact that we have this bug is a strong argument against using `Integer` to represent an optional integer in `MappingConfig`.  Also this is _also_ not correct, just in a different way: `intValue` will fail when `getPriority` is `null`.

So AFAICS we could _anyway_ keep priority as an `int` (maybe with a magic value -1 for "default").

Reply:
I'll do a mix of the two: the constructor will still receive an `Integer` as an argument so we can check for `null` values instead of the client classes passing the default number. But the class's field type will be `int` and it will be initialized by default to -1 and will get changed if the passed arg isn't null

Reply:
Maybe just 2 constructors, then?